src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue

```diff
@@ -0,0 +1,233 @@
+<script setup>
+import { ref, onMounted, watch, computed } from "vue";
+import { useRetryEditedMessage } from "../../composables/serviceFailedMessage";
+import MessageHeader from "./EditMessageHeader.vue";
+
+const emit = defineEmits(["cancel", "retried"]);
+
+const settings = defineProps({
+  id: String,
+  message: Object,
+  configuration: Object,
+});
+
+let panel = ref();
+let message = ref();
+let origMessageBody = undefined;
+
+let showEditAndRetryConfirmation = ref(false);
+let showCancelConfirmation = ref(false);
+let showEditRetryGenericError = ref(false);
+
+const id = computed(() => settings.id);
+const messageBody = computed(() => settings.message.messageBody);
+
+watch(messageBody, async (newValue) => {
+  if (newValue !== origMessageBody) {
+    message.value.isBodyChanged = true;
+  }
+  if (newValue === "") {
+    message.value.isBodyEmpty = true;
+  } else {
+    message.value.isBodyEmpty = false;
+  }
+});
+
+function close() {
+  emit("cancel");
+}
+
+function confirmEditAndRetry() {
+  showEditAndRetryConfirmation.value = true;
+}
+
+function confirmCancel() {
+  if (message.value.isBodyChanged) {
+    showCancelConfirmation.value = true;
+    return;
+  }
+
+  if (message.value.headers.find((header) => header.isChanged === true)) {
+    showCancelConfirmation.value = true;
+    return;
+  }
+
+  close();
+}
+
+function resetBodyChanges() {
+  message.value.messageBody = origMessageBody;
+  message.value.isBodyChanged = false;
+}
+
+function findHeadersByKey(key) {
+  return message.value.headers.find((header) => header.key === key);
+}
+
+function getContentType() {
+  var header = findHeadersByKey("NServiceBus.ContentType");
+  return header ? header.value : null;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

```suggestion
  return header?.value;
```